pybedlite/overlap_detector.py

```diff
@@ -100,6 +102,49 @@ def length(self) -> int:
         """Returns the length of the interval."""
         return self.end - self.start
 
+    def to_bedrecord(self) -> BedRecord:
+        """
+        Convert an `Interval` to a `BedRecord` instance.
+
+        **Note that `Interval` cannot represent a `BedRecord` with a missing strand.**
+        Converting a record with no strand to `Interval` and then back to `BedRecord` will result in
+        a record with **positive strand**.
+
+        Returns:
+            An `Interval` corresponding to the same region specified in the record.
+        """
+
+        return BedRecord(
+            chrom=self.refname,
+            start=self.start,
+            end=self.end,
+            strand=BedStrand.Negative if self.negative else BedStrand.Positive,
+            name=self.name,
+        )
+
+    @classmethod
+    def from_bedrecord(cls: Type["Interval"], record: BedRecord) -> "Interval":
```

Comment:
I'd put this into `BedRecord` as `to_interval`

Reply:
Perhaps we need to standardize to either `to*` or `from*` like we do in Scala.

Reply:
I am always in favor of standardization!

I have no preference between the two. However, I placed both of these on `Interval` because importing `Interval` into `bed_record` creates a circular import.

- `bed_record` imports `Interval` from `overlap_detector`
- `overlap_detector` imports `BedSource` from `bed_source`
- `bed_source` imports `BedRecord` from `bed_record`

I think we would need to move `Interval` to a separate module but I also think that would break existing imports of this package

Would love to be wrong though :)

Reply:
I standardized to `from_FOO()` since imo alternative constructors should be classmethods on the class they construct. (And if we did `to_FOO()`, we'd truly run into a circular import since we'd have to instantiate `Interval` and `BedRecord` in each other's modules)